core-plugins/src/main/java/io/cdap/plugin/transform/JavaScriptTransform.java

```diff
@@ -196,7 +197,8 @@ void setErrorSchema(Schema schema) {
   public void transform(StructuredRecord input, Emitter<StructuredRecord> emitter) {
     try {
       engine.eval(String.format("var %s = %s;", VARIABLE_NAME, GSON.toJson(input)));
-      Emitter<Map> jsEmitter = new JSEmitter(emitter, schema == null ? input.getSchema() : schema);
+      Emitter<Map> jsEmitter = new JSEmitter(emitter,
```

Comment:
we should not be changing transform method. I think the changes in init method should be sufficient.

Reply:
Fixed